lockfile/src/lib.rs

```diff
@@ -372,9 +372,8 @@ pub fn find_depfiles_at(root: impl AsRef<Path>) -> Vec<(PathBuf, LockfileFormat)
             // Legacy Gradle (before v7) lockfiles are in a subdirectory,
             // so we truncate these directories to get the effective
             // directory these lockfiles were created for.
-            let dir_str = lockfile_dir.to_string_lossy();
             if lockfile_format == &LockfileFormat::Gradle
-                && dir_str.ends_with("/gradle/dependency-locks")
+                && lockfile_dir.ends_with("gradle/dependency-locks")
```

Comment:
I wasn't aware that this would actually work considering the following statement in Rust's docs:

> Only considers whole path components to match.

But I've tested it and this does indeed work. While it doesn't behave as I would expect, it does behave like we want here. The path **must** include both components in their entirety.

The only notable thing is that `/path/gradle/dependency-locks/` also matches, so both with and without the trailing slash matches. But I don't think this should be an issue in our case, since we're matching on a directory anyway.